### DISCUSSION TITLE
仿真策略正常运行，但是无法进入on_calculate

### DISCUSSION THREAD

**Discussion by delusionpig** [wondertrader/wondertrader#54] [Q&A] [answered]
问题描述：CTA引擎下策略，回测正常运行，切换到仿真使用simnow进行仿真时，m1和m5频率下策略没有回调on_calculate,但是显示了回调了on_tick。

on_calculate\on_bar\on_tick分别添加了print，最终输出的只有on_tick执行了，其他一切正常，数据能够获取到，历史数据和实时数据都有

![image](https://user-images.githubusercontent.com/91172056/162108385-8452148f-70ee-4ce0-8a1e-9c2f41548684.png)
![image](https://user-images.githubusercontent.com/91172056/162108468-16eba299-a785-4396-9f34-e9d899240c71.png)
![image](https://user-images.githubusercontent.com/91172056/162108494-bd5281b9-5549-4e34-8a46-a838852c6588.png)

log：
![image](https://user-images.githubusercontent.com/91172056/162109253-d24a2e27-359e-4fe7-85bd-53217c61765a.png)

休市期间异常情况：
![image](https://user-images.githubusercontent.com/91172056/162109325-5cc2a419-a83a-4897-9f25-cf3f0dd037b7.png)

**Comment by delusionpig** [comment, marked as answer]
正常交易时间一直没有调用on_calculate 和 on_bar,10:25的时候又进去了

**Reply by wondertrader** [reply]
这个问题还有吗

**Reply by delusionpig** [reply]
pip install wtpy -U之后就解决了

**Reply by delusionpig** [reply]
该问题应该是由于数据接收导致的，如果tick数据没有接受，就会导致k线没有闭合，导致onbar oncal无法进入，开盘前重启datakit和策略即可
